homeassistant/components/econet/climate.py

```diff
@@ -0,0 +1,243 @@
+        super().__init__(thermostat)
+        self._running = thermostat.running
+        self._poll = True
+        self.thermostat = thermostat
+        self.econet_state_to_ha = {}
+        self.ha_state_to_econet = {}
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        if self.thermostat.supports_humidifier:
+            return SUPPORT_FLAGS_THERMOSTAT | SUPPORT_TARGET_HUMIDITY
+        return SUPPORT_FLAGS_THERMOSTAT
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self.thermostat.set_point
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        return self.thermostat.humidity
+
+    @property
+    def target_humidity(self):
+        """Return the humidity we try to reach."""
+        if self.thermostat.supports_humidifier:
+            return self.thermostat.dehumidifier_set_point
+        return None
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        if self.hvac_mode == HVAC_MODE_COOL:
+            return self.thermostat.cool_set_point
+        if self.hvac_mode == HVAC_MODE_HEAT:
+            return self.thermostat.heat_set_point
+        return None
+
+    @property
+    def target_temperature_low(self):
+        """Return the lower bound temperature we try to reach."""
+        if self.hvac_mode == HVAC_MODE_AUTO:
+            return self.thermostat.heat_set_point
+        return None
+
+    @property
+    def target_temperature_high(self):
+        """Return the higher bound temperature we try to reach."""
+        if self.hvac_mode == HVAC_MODE_AUTO:
+            return self.thermostat.cool_set_point
+        return None
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        target_temp = kwargs.get(ATTR_TEMPERATURE)
+        target_temp_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
+        target_temp_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+        if target_temp:
+            self.thermostat.set_set_point(target_temp, None, None)
+            return
+        if target_temp_low or target_temp_high:
+            self.thermostat.set_set_point(None, target_temp_high, target_temp_low)
+            return
+
+    @property
+    def is_aux_heat(self):
+        """Return true if aux heater."""
+        return self.thermostat.mode == ThermostatOperationMode.EMERGENCY_HEAT
+
+    @property
+    def hvac_modes(self):
+        """Return hvac operation ie. heat, cool mode.
+
+        Needs to be one of HVAC_MODE_*.
+        """
+        econet_modes = self.thermostat.modes
+        op_list = []
+        for mode in econet_modes:
+            if mode not in [
+                ThermostatOperationMode.UNKNOWN,
+                ThermostatOperationMode.EMERGENCY_HEAT,
+            ]:
+                ha_mode = ECONET_STATE_TO_HA[mode]
+                op_list.append(ha_mode)
```

Comment:
Canw ec slculate this list in init method and save it as an instance attribute?

Reply:
Makes sense